### ISSUE TITLE
Printer improperly quotes symbols containing semicolons (and others)

### ISSUE THREAD

**Issue by dvanhorn** [racket/racket#3659]
The default printer doesn't properly quote symbols containing `;` such as `'|;|`, which is inconsistent with the handling of other characters that need to be quoted in symbols, e.g. `'| |`.  It drops the vertical bars, resulting in something that can't be read back as the symbol itself: `';`.

I've noticed the same issue for `'|'|`, `(string->symbol (string (integer->char 96)))` (I would write this more directly, but it seems GH markdown won't allow it 😄), but not `'|,|` or `'|,@|'`).  

**What version of Racket are you using?**
7.9 [cs]

**What program did you run?**
```
Welcome to Racket v7.9 [cs].
> '|;|
';
```

**What should have happened?**
```
Welcome to Racket v7.9 [cs].
> '|;|
'|;|
```

**Comment by jackfirth** [comment]
Does it work on BC?

**Comment by mflatt** [comment]
This has been fixed for v8.0 in 3b34b0ce02d5995dce7372665706a5f3ec6b0f0a, at least for semicolon and backquote. I'm not able to replicate the problem for quote.

**Comment by dvanhorn** [comment]
Great, thanks.  I was wrong about quote, it seems to print fine.  And this doesn't happen on 7.9 BC, so I'll close.
